src/queueUntilType.js

```diff
@@ -0,0 +1,35 @@
+import { subscribeToResult } from 'rxjs/util/subscribeToResult';
+
+export function queueUntilType(expectedType) {
+  return this.lift(function(source) {
+    let queue = [];
+    let hasSeenType = false;
+    const nullifyQueue = () => { queue = null; };
+
+    const outer = {
+      notifyNext: (_, action) => this.next(action),
+      notifyComplete: nullifyQueue
+    };
+    return source.subscribe({
+      next: action => {
+        // It's important to check this first because of possible re-entry
+        // before we've finished flushing the queue.
+        if (hasSeenType) {
+          return this.next(action);
+        }
+
+        if (action && action.type === expectedType) {
+          hasSeenType = true;
+          this.add(subscribeToResult(outer, queue));
+          // no longer needed, free up memory
+          nullifyQueue();
+        } else {
+          queue.push(action);
+        }
+      },
+      error: e => this.error(e),
+      complete: () => this.complete(),
+      unsubscribe: nullifyQueue
```

Comment:
it was just AFAIK, so AFAIK passing unsubscribe like that does nothing at all. I think the object will get passed to `new Subscriber(yourObserver)` which will internally wrap it in `SafeSubscriber` neither of which (again AFAICT) call unsubscribe on any passed in observer object. Quick testing suggests this as well: https://jsbin.com/dozuseg/edit?js,console

Certainly I _could be wrong_ as I have many times hehe 😈 If I am, this would be very interesting to know so lmk if you can reproduce.